core/services/relay/evm/mercury/wsrpc/client.go

```diff
@@ -119,14 +155,21 @@ func newClient(lggr logger.Logger, clientPrivKey csakey.KeyV2, serverPubKey []by
 		dialSuccessCountMetric:     dialSuccessCount.WithLabelValues(serverURL),
 		dialErrorCountMetric:       dialErrorCount.WithLabelValues(serverURL),
 		connectionResetCountMetric: connectionResetCount.WithLabelValues(serverURL),
+		tlsCertFile:                tlsCertFile,
 	}
 }
 
 func (w *client) Start(ctx context.Context) error {
-	return w.StartOnce("WSRPC Client", func() (err error) {
+	name := "WSRPC Client"
+	if w.tlsCertFile != nil {
+		name = "GRPC Client"
+	}
+
+	return w.StartOnce(name, func() (err error) {
+		// TODO: this is unecessarily restrictive, the consumer can decide.
 		// NOTE: This is not a mistake, dial is non-blocking so it should use a
 		// background context, not the Start context
-		if err = w.dial(context.Background()); err != nil {
+		if err = w.chooseDial(ctx); err != nil {
```

Comment:
```
// NOTE: This is not a mistake, dial is non-blocking so it should use a
// background context, not the Start context
```

This comment is no longer in agreement with the code. Was it intentional to replace `context.Background()` with Start context ?

Reply:
My intended approach is from my comment - I think its restrictive to insert a background context; the consumer can decide.